src/components/ComputeModal.js

```diff
@@ -1468,18 +1468,18 @@ export const ComputeModalBase = ({
               'This change requires rebuilding your cloud environment, which will ',
               span({ style: { fontWeight: 600 } }, ['delete all files on built-in hard disk.'])
             ]),
-            existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
+            existingRuntime?.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willDeletePersistentDisk(), () => h(Fragment, [
             p([
               'To reduce the size of the PD, the existing PD will be deleted and a new one will be created and attached to your virtual machine instance. This will ',
               span({ style: { fontWeight: 600 } }, ['delete all files on the disk.'])
             ]),
-            existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
+            existingRuntime?.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willRequireDowntime(), () => h(Fragment, [
-            existingRuntime.tool !== desiredTool ?
-              p(['By continuing, you will be changing the application of your cloud environment from ', strong([existingRuntime.tool]), ' to ',
+            existingRuntime && existingRuntime.tool !== desiredTool ?
+              p(['By continuing, you will be changing the application of your cloud environment from ', strong([existingRuntime?.tool]), ' to ',
```

Comment:
This is in a branch that's conditional on `existingRuntime` being truthy, so optional chaining is not necessary.

Reply:
Yes, but it seems code written not-defensively eventually gets refactored to a place where it bites us... I'd prefer to be overly cautious here.

Reply:
I would argue that this is not defensive coding. Maybe it prevents a JS error (reading a property of undefined), but if `existingRuntime` is undefined here, then the output is still _incorrect_. It's an error in a different form. Given the choice between the two types of errors, I'd claim that it's preferable to fail loudly since it reduces the chances that the problem will go undetected during development or testing.